pkg/controller/kanarydeployment/strategies/factory.go

```diff
@@ -79,10 +80,17 @@ func (s *strategy) Apply(kclient client.Client, reqLogger logr.Logger, kd *kanar
 	var newStatus *kanaryv1alpha1.KanaryDeploymentStatus
 	newStatus, result, err = s.process(kclient, reqLogger, kd, dep, canarydep)
 	utils.UpdateKanaryDeploymentStatusConditionsFailure(newStatus, metav1.Now(), err)
-	return utils.UpdateKanaryDeploymentStatus(kclient, reqLogger, kd, newStatus, result, err)
+	if os.Getenv("SUBRESOURCE_DISABLED") == "1" {
```

Comment:
It can be very expensive to call each time `os.Getenv(...)` we should sort the value in the 'strategy'

Reply:
fixed